misc/lib-creator/src/main/java/org/ballerinalang/stdlib/utils/GenerateBalo.java

```diff
@@ -159,23 +168,23 @@ private static void genApiDoc(String sourceRootDir, Set<String> docModuleFilter,
         BallerinaDocGenerator.writeAPIDocsForModules(moduleDocMap, apiDocPath, false);
     }
 
-    private static void printErrors(boolean reportWarnings, CompileResult.CompileResultDiagnosticListener diagListner,
+    private static void printErrors(boolean reportWarnings,
                                     List<Diagnostic> diagnostics, boolean logIsolationWarnings) {
         int deprecatedWarnCount = 0;
-        if (reportWarnings && diagListner.getWarnCount() > 0) {
-            for (Diagnostic diagnostic : diagListner.getDiagnostics()) {
-                DiagnosticCode code = diagnostic.getCode();
+        if (reportWarnings && warnCount > 0) {
+            for (Diagnostic diagnostic : diagnostics) {
+                DiagnosticCode code = ((BLangDiagnostic) diagnostic).getCode();
                 if (code == USAGE_OF_DEPRECATED_CONSTRUCT || (!logIsolationWarnings && isIsolatedWarningLog(code))) {
```

Comment:
`code` can be null. That would result in a NPE inside the `isIsolatedWarningLog()` method